python_modules/dagster/dagster/_core/definitions/metadata/source_code.py

```diff
@@ -0,0 +1,154 @@
+import inspect
+import os
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Mapping,
+    Optional,
+    Sequence,
+    Union,
+)
+
+import pydantic
+
+import dagster._check as check
+from dagster._annotations import experimental
+from dagster._model import DagsterModel
+from dagster._serdes import whitelist_for_serdes
+
+from .metadata_set import (
+    NamespacedMetadataSet as NamespacedMetadataSet,
+    TableMetadataSet as TableMetadataSet,
+)
+from .metadata_value import MetadataValue
+
+if TYPE_CHECKING:
+    from dagster._core.definitions.assets import AssetsDefinition, SourceAsset
+    from dagster._core.definitions.cacheable_assets import CacheableAssetsDefinition
+
+DEFAULT_SOURCE_FILE_KEY = "asset_definition"
+
+
+@experimental
+@whitelist_for_serdes
+class LocalFileSource(DagsterModel):
+    """Represents a local file source location."""
+
+    file_path: str
+    line_number: int
+
+
+@experimental
+@whitelist_for_serdes
+class SouceCodeLocationsMetadataValue(
```

Comment:
In isolation, I think "source code location" is a solid name for this concept, but "code location" already means something in our system which is pretty different from what this means. That makes me think we should find a less overlapping name.

@schrockn has also been pushing for us to avoid class names with plurals, so something like "XSet" would be preferable to "Xs".

Reply:
Way more comfortable with a plural as an adjective here because any plausible variable/symbol name would end with `_value`

Reply:
Share concern about source code location.

What about just `SourceLocation` or `SourceTextLocation`, or `SourceFileLocation`?

Reply:
> Way more comfortable with a plural as an adjective here because any plausible variable/symbol name would end with _value

👍 

> What about just SourceLocation or SourceTextLocation, or SourceFileLocation?

I think we need to be careful about collisions with future names that refer to the location of source data consumed by a pipeline.